skimage/segmentation/_join.py

```diff
@@ -56,7 +56,7 @@ def join_segmentations(s1, s2, return_mapping: bool = False):
     s1_relabeled, _, backward_map1 = relabel_sequential(s1)
     s2_relabeled, _, backward_map2 = relabel_sequential(s2)
     # Create joined label image
-    factor = s2.max() + 1
+    factor = (s2.max() + 1).astype(s2.dtype)
```

Comment:
```suggestion
    factor = (s2.max() + 1).astype(s2.dtype, copy=False)
```
Avoids unnecessary copies.